packages/router/router.ts

```diff
@@ -3038,6 +3114,73 @@ function shouldRevalidateLoader(
   return arg.defaultShouldRevalidate;
 }
 
+/**
+ * Execute route.lazy() methods to lazily load route modules (loader, action,
+ * shouldRevalidate) and update the routeManifest in place which shares objects
+ * with dataRoutes so those get updated as well.
+ */
```

Comment:
Definitely open to feedback on this function - left some longer comment sin here about it's current state, but this probably could use some battle testing to make sure the assumptions are valid.